src/raftstore/store/store.rs

```diff
@@ -1918,19 +1918,23 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         };
 
         let region = peer.region();
+        let latest_epoch = region.get_region_epoch();
 
-        if region.get_region_epoch().get_version() != epoch.get_version() {
+        if latest_epoch.get_version() != epoch.get_version() {
             info!(
-                "{} epoch changed {:?} != {:?}, need re-check later",
+                "{} epoch changed {:?} != {:?}, retry later",
                 peer.tag,
                 region.get_region_epoch(),
                 epoch
             );
-            return Err(box_err!(
-                "{} epoch changed {:?} != {:?}, need re-check later",
-                peer.tag,
-                region.get_region_epoch(),
-                epoch
+            return Err(Error::StaleEpoch(
```

Comment:
why can return StaleEpoch fix the problem?

Reply:
https://github.com/pingcap/tikv/blob/d780a5f9ba18cdb2a7063172079ff9bfe27a06d4/tests/raftstore/cluster.rs#L762-L769

The line 768 panics if the respones has a box error, however it will retry, if the error is the stale epoch.